From a803fa6605ad70731081942d50f1fc542a499110 Mon Sep 17 00:00:00 2001
From: Zhen Chen <chenzhen@rock-chips.com>
Date: Fri, 15 Nov 2019 21:09:57 +0800
Subject: [PATCH] Mali: utgard: fix memory(fences) leaks

In deed, Zhixiong Lin missed these codes when he picking source codes
from DDK r9 to port utgard device driver of r7 to kernel 4.19.

Change-Id: If140b478b4ea8a4f72d532b8e8216ab53856be08
Signed-off-by: Zhen Chen <chenzhen@rock-chips.com>
---
 drivers/gpu/arm/mali400/mali/linux/mali_sync.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/arm/mali400/mali/linux/mali_sync.c b/drivers/gpu/arm/mali400/mali/linux/mali_sync.c
index 6074ac3f5484..0d98b518f1ac 100755
--- a/drivers/gpu/arm/mali400/mali/linux/mali_sync.c
+++ b/drivers/gpu/arm/mali400/mali/linux/mali_sync.c
@@ -612,6 +612,10 @@ struct mali_internal_sync_fence *mali_sync_flag_create_fence(struct mali_sync_fl
 		return NULL;
 	}
 
+	/* 'sync_pt' no longer needs to hold a refcount of '*sync_pt', to put it off. */
+	dma_fence_put(&sync_pt->base);
+	sync_pt = NULL;
+
 	return sync_fence;
 }
 #endif
-- 
2.35.3

